Lib/asyncio/base_events.py

```diff
@@ -458,24 +458,26 @@ def create_future(self):
         """Create a Future object attached to the loop."""
         return futures.Future(loop=self)
 
-    def create_task(self, coro, *, name=None, context=None):
+    def create_task(self, coro, *, name=None, context=None, eager_start=None):
         """Schedule a coroutine object.
 
         Return a task object.
         """
         self._check_closed()
         if self._task_factory is None:
-            task = tasks.Task(coro, loop=self, name=name, context=context)
+            task = tasks.Task(coro, loop=self, name=name, context=context, eager_start=eager_start)
             if task._source_traceback:
                 del task._source_traceback[-1]
         else:
-            if context is None:
+            if context is None and eager_start is None:
                 # Use legacy API if context is not needed
                 task = self._task_factory(self, coro)
-            else:
+                task.set_name(name)
+            elif eager_start is None:
                 task = self._task_factory(self, coro, context=context)
-
-            task.set_name(name)
+                task.set_name(name)
```

Comment:
this set_name is too late because the task could have already run - so it needs a bit of thought

Reply:
https://github.com/python/cpython/issues/128308

Reply:
We can check first if the task factory exactly is `eager_task_factory`, and call it with all provided arguments.
The last resort for unknown task factory could have flaws.

On the other side, we can use `inspect.signature()` + some form of cache to analyze in runtime what arguments are accepted. I'm not a big fan of this approach, but maybe we have not choice.
Support of third-party tasks and factories adds a mess; we should live with it.

Reply:
I plan to deprecate the eager_start=None branches, so perhaps this is not so bad